openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,54 +432,30 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
+Note: To prevent downgrade attacks, a Client can choose to include certain parameters in the request object or not.
 
-To use `client_id_scheme` values `entity_id`, `did`, `verifier_attestation`, `x509_san_dns`, and `x509_san_uri`, Verifiers MUST be confidential clients. This might require changes to the technical design of native apps as such apps are typically public clients.
+The following is a list of methods defined by this specification that MAY be used by the Wallet to authenticate the Request Object against the `client_id` in the Request Object and verify the `redirect_uri` or `response_uri` belong to the Client:
+- Pre-registered: If the Client is pre-registered, the `jwks` Client Metadata parameter provided upon registration MUST be used to verify the signature of the Request Object.
+- X.509 certificates: If the Request Object contains an `x5c`, `x5t`, or `x5t#S256` header parameter, after validating the X.509 certificate chain, the Wallet MUST use the public key from the leaf-certificate of the X.509 certificate chain to verify the signature of the Request Object. In this case, the leaf-certificate MUST include a`dNSName` Subject Alternative Name (SAN) [RFC5280] that matches the FQDN of the `client_id`, or `uniformResourceIdentifier` SAN [RFC5280] that matches the `client_id`. If the Wallet can establish trust in the `client_id` authenticated through the certificate, e.g. because the `client_id` is contained in a list of trusted Client Identifiers, it MAY allow the Client to freely choose the `redirect_uri` or `response_uri` value. If not, the `redirect_uri` or `response_uri` value MUST match the `client_id`.
+- DID Resolution: If the `client_id` in the Request Object is a DID, a public key to verify the signature MUST be obtained from the `verificationMethod` property of a DID Document. Since a DID Document can include multiple public keys, a particular public key used to sign the request in question MUST be identified by the `kid` header parameter. To obtain the DID Document, the Wallet MUST use DID Resolution defined by the DID method used by the Verifier.
```

Comment:
```suggestion
- DID Resolution: If the `client_id` in the Request Object is a DID, a public key to verify the signature MUST be obtained from a `verificationMethod` property of a DID Document. Since a DID Document can include multiple public keys, the particular public key used to sign the request MUST be identified by the `kid` header parameter. To obtain the DID Document, the Wallet MUST use DID Resolution as defined by the DID method used by the Verifier.
```